src/include/duckdb/main/config.hpp

```diff
@@ -246,6 +249,8 @@ struct DBConfig {
 	shared_ptr<BufferManager> buffer_manager;
 	//! Set of callbacks that can be installed by extensions
 	vector<unique_ptr<ExtensionCallback>> extension_callbacks;
+	//! AES state for en/decryption
+	shared_ptr<AESStateFactory> encryption_state;
```

Comment:
Perhaps we can rename `AESStateFactory` to e.g. `EncryptionUtil` to be more generic/allow for additional expansion here in the future?